src/plugins/gs-plugin-external-appstream.c

```diff
@@ -109,8 +152,50 @@ update_external_appstream (GsPlugin *plugin,
 	if (!create_tmp_file (tmp_file_name, &tmp_file, error))
 		return FALSE;
 
-	if (!gs_plugin_download_file (plugin, NULL, url, tmp_file, cancellable,
-				      error)) {
+	msg = soup_message_new (SOUP_METHOD_GET, url);
+
+	/* Set the If-Modified-Since header if the target file exists */
+	target_file_path = g_build_filename (APPSTREAM_SYSTEM_DIR, file_name,
+					     NULL);
+	local_mod_date = get_modification_date (target_file_path);
```

Comment:
I wonder if this "check modification date" functionality might be better in gs-utils.c or  gs-plugin.c -- it seems useful enough that other plugins would want to use it.